src/Tasks/Exec.cs

```diff
@@ -203,7 +198,7 @@ public ITaskItem[] Outputs
         /// </summary>
         private void CreateTemporaryBatchFile()
         {
-            var encoding = BatchFileEncoding();
+            var encoding = EncodingUtilities.BatchFileEncoding(Command + WorkingDirectory, UseUtf8Encoding);
 
             // Temporary file with the extension .Exec.bat
             _batchFile = FileUtilities.GetTemporaryFile(".exec.cmd");
```

Comment:
Unrelated to this PR but FYI: From what I learned from the console team, the code slightly below those lines isn't how things should work "in an ideal world" (not sure how this is actually feasible)

```
_standardOutputEncoding = encoding;
_standardErrorEncoding = encoding;
```

The fact that we use chcp to change the console code page and console output code page to some arbitrary code page shouldn't affect the encoding of the stdout written by well-behaved apps (again, according to the console team's ideal world). Because the output is redirected to msbuild and not sent to a console buffer, apps should always write UTF-8 in order to preserve all unicode chars. Apps should only honor the console output code page when writing to a conhost's buffer. This differs from what I previously thought was the preferred behavior.